.github/scripts/run-container-test.py

```diff
@@ -0,0 +1,145 @@
+
+This requires Docker multiarch support, see https://docs.docker.com/build/building/multi-platform/ and https://github.com/multiarch/qemu-user-static
+In GitHub we use a provided action for this: https://github.com/docker/setup-qemu-action
+
+Locally you would need to run one of:
+
+`docker run --rm --privileged multiarch/qemu-user-static --reset -p yes --credential yes`
+
+OR
+
+`docker run --privileged --rm tonistiigi/binfmt --install all`
+"""
+
+import argparse
+import os
+import subprocess
+import shlex
+import shutil
+
+VERBOSE = False
+
+DISTRO_TO_IMAGE_NAME= {
+    "ubuntu-22.04": "public.ecr.aws/lts/ubuntu:22.04_stable",
+    "al2023": "public.ecr.aws/amazonlinux/amazonlinux:2023",
+    "al2": "public.ecr.aws/amazonlinux/amazonlinux:2"
+}
+
+DOCKER_PLATFORM_BY_ARCH = {
+    "x64": "linux/amd64",
+    "arm64": "linux/arm64"
+}
+
+def vprint(message):
+    global VERBOSE
+    if VERBOSE:
+        print(message)
+
+
+def running_in_github_action():
+    """
+    Test if currently running in a GitHub action or running locally
+    :return: True if running in GH, False otherwise
+    """
+    return "GITHUB_WORKFLOW" in os.environ
+
+
+def shell(command, cwd=None, check=True, capture_output=False):
+    """
+    Run a command
+    :param command: command to run
+    :param cwd: the current working directory to change to before executing the command
+    :param check: flag indicating if the status code should be checked. When true an exception will be
+    thrown if the command exits with a non-zero exit status.
+    :returns: the subprocess CompletedProcess output
+    """
+    # if type(command) is not list:
+    #     command = shlex.split(command)
+
+    vprint(f"running `{command}`")
+    return subprocess.run(command, shell=True, check=check, cwd=cwd, capture_output=capture_output)
+
+
+def oci_executable():
+    """
+    Attempt to find the OCI container executor used to build and run docker containers
+    """
+    oci_exe = os.environ.get('OCI_EXE')
+    if oci_exe is not None:
+        return oci_exe
+    
+    executors = ['finch', 'podman', 'docker']
+
+    for exe in executors:
+        if shutil.which(exe) is not None:
+            return exe
+
+    print("cannot find container executor")
+    exit(1)
+
+
+def run_docker_test(opts): 
+    """
+    Run a docker test for a precompiled Kotlin/Native binary
+
+    :param opts: the parsed command line options
+    """
+    platform = DOCKER_PLATFORM_BY_ARCH[opts.arch]
+    oci_exe = oci_executable()
+
+    test_bin_dir = os.path.abspath(opts.test_bin_dir)
+    image_name = DISTRO_TO_IMAGE_NAME[opts.distro]
+    path_to_exe = f'./linux{opts.arch.capitalize()}/debugTest/test.kexe'
+
+    cmd = [
+        oci_exe,
+        'run',
+        '--rm',
+        f'-v{test_bin_dir}:/test',
+        '-w/test',
+        '-e DEBIAN_FRONTEND=noninteractive',
+        '--platform',
+        platform,
+        image_name,
+        path_to_exe,
+    ]
+
+    cmd = shlex.join(cmd)
+    print(cmd)
```

Comment:
question: is this `print` intentionally left here? should it be `vprint` instead?

Reply:
I intentionally left it, I could probably just remove `vprint`. This was cobbled together from existing scripts but I left it in since it's basic building blocks should we add more to it.